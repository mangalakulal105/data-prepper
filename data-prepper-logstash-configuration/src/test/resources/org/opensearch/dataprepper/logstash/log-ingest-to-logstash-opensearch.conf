input {
    http {
    }
}
filter {
    csv {
        autogenerate_column_names => true
        columns => ["a", "b"]
        quote_char => ";"
        separator => ":"
        source => "message_source"
    }
    grok {
        match => {"log" => "%{COMBINEDAPACHELOG}"}
    }
    date {
        match => ["timestamp", "yyyy-MM-dd HH:mm:ss", "yyyy MMM dd HH:mm"]
        target => "@timestamp"
        locale => "en_US"
        timezone => "UTC"
    }
}
output {
    opensearch {
        hosts => ["fakedomain.us-east-1.es.amazonaws.com"]
        user => myuser
        password => mypassword
        index => "my-index"
        ssl_certificate_verification => true
    }
}