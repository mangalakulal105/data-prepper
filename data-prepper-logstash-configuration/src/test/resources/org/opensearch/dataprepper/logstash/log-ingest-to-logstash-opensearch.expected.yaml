logstash-converted-pipeline:
  source:
    http:
      max_connection_count: 500
      request_timeout: 10000
  processor:
    - csv:
        delete_header: true
        column_names:
          - "a"
          - "b"
        quote_character: ";"
        delimiter: ":"
        source: "message_source"
    - grok:
        match:
          log:
            - "%{COMBINEDAPACHELOG}"
    - date:
        from_time_received: false
        destination_timezone: "UTC"
        destination: "@timestamp"
        locale: "en_US"
        source_timezone: "UTC"
        match:
          - key: "timestamp"
            patterns:
              - "yyyy-MM-dd HH:mm:ss"
              - "yyyy MMM dd HH:mm"
  sink:
    - opensearch:
        hosts:
          - "fakedomain.us-east-1.es.amazonaws.com"
        username: "myuser"
        password: "mypassword"
        insecure: false
        index: "my-index"
